src/api/goods.ts

```
@@ -3,14 +3,35 @@
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+export async function getAll(): Promise<Good[]> {
+  try {
+    const response = await fetch(API_URL);
+    if (!response.ok) {
+      throw new Error(`HTTP Error: ${response.status}`);
+    }
+
+    return response.json();
+  } catch (error) {
+    return [];
+  }
 }
 
-export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+export const get5First = async () => {
+  try {
+    const goods = await getAll();
+    return [...goods]
+      .sort((good1: Good, good2: Good) => good1.name.localeCompare(good2.name))
+      .slice(0, 5);
+  } catch (error) {
+    return [];
+  }
 };
 
-export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
+export const getRedGoods = async () => {
+  try {
+    const goods = await getAll();
+    return goods.filter((good: Good) => good.color === 'red');
+  } catch (error) {
+    return [];
```

Catching errors here and returning `[]` hides upstream failures from the UI/tests. This violates the checklist requirement: "Catch and handle network or parsing errors during fetch. Propagate or display the error state in a way that can be verified in code (e.g., setting error state and conditional rendering) (see checklist item 3)." Propagate the error (rethrow) so the application can show the error state, rather than returning an empty list.